include/plugins/audioplayers.h

```diff
@@ -0,0 +1,53 @@
+#ifndef AUDIOPLAYERS_H_
+#define AUDIOPLAYERS_H_
+
+#include <gst/gst.h>
+#include <stdbool.h>
+
+typedef struct AudioPlayer
+{
+    GstElement* playbin;
+    GstElement* source;
+    GstBus* bus;
+
+    bool _isInitialized;
+    bool _isLooping;
+    bool _isSeekCompleted;
+    double _playbackRate;
+
+    char* url;
+    char* playerId;
+    /* FlMethodChannel* _channel; */
+    char* _channel;
+
+} AudioPlayer;
```

Comment:
```suggestion
struct audio_player;
```

and then somewhere in audioplayers.c:
```C
struct audio_player {
    GstElement *playbin;
    GstElement *source;
    GstBus *bus;

    bool is_initialized;
    bool is_looping;
    bool is_seek_completed;
    double playback_rate;

    char *url;
    char *player_id;
    char *channel;
};
```

Whether to use typedefs in C is a whole topic on it's own, but I chose not to use them in flutter-pi so let's not deviate here. But at least the implementation of `audio_player` should be hidden to other files.

```C
// Also, align the `*` of a pointer declaration to the variable name rather than the type, like this:
void *ptr;
// Rather than this:
void* ptr;

// In C, the ptr aspect `*` is not part of the type but of the declared variable, i.e.
int myint, *myintptr;
// Would declare an int and a pointer-to-int
```

any names should be `lower_camel_case` (exception is enums if you use them).

Opening `{` should be on the same line, not on a separate one

I have a clang-format file you can use that fixes most stuff, haven't applied it to the source yet because that means merging `feature/compositor-ng` into `master` will be painful with the thousands of whitespace changes.